Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/NodeActionListener.java

```diff
@@ -159,4 +164,34 @@ protected void afterActionPerformed(NodeActionEvent e) {
         // mark node as done loading
     }
 
+    public final NodeActionListener asGenericListener() {
+        if (this instanceof DelegateActionListener) {
+            return this;
+        }
+        NodeActionListener delegate = this;
+        if (this instanceof Backgroundable) {
```

Comment:
this may implements multi interface, or NodeActionListener should only implement one of below interface?

Reply:
yes. it could.

Reply:
If so, `delegate` will be replaced if `NodeActionListener` implement multi interfaces

Reply:
yes. that's we expected.